src/Agent.Worker/TestResults/TestRunPublisher.cs

```diff
@@ -0,0 +1,306 @@
+                    _testRun.Id);
+                trTask.Wait();
+            }
+        }
+
+        private String GetAttachmentType(String file)
+        {
+            string fileName = Path.GetFileNameWithoutExtension(file);
+
+            if (String.Compare(Path.GetExtension(file), ".coverage", StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return AttachmentType.CodeCoverage.ToString();
+            }
+            else if (String.Compare(Path.GetExtension(file), ".trx", StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return AttachmentType.TmiTestRunSummary.ToString();
+            }
+            else if (String.Compare(fileName, "testimpact", StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return AttachmentType.TestImpactDetails.ToString();
+            }
+            else if (String.Compare(fileName, "SystemInformation", StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return AttachmentType.IntermediateCollectorData.ToString();
+            }
+            else
+            {
+                return AttachmentType.GeneralAttachment.ToString();
+            }
+        }
+
+        private TestAttachmentRequestModel GetAttachmentRequestModel(string attachment)
+        {
+            if (File.Exists(attachment) && new FileInfo(attachment).Length <= TCM_MAX_FILESIZE)
+            {
+                byte[] bytes = File.ReadAllBytes(attachment);
+                string encodedData = Convert.ToBase64String(bytes);
+                if (encodedData.Length <= TCM_MAX_FILESIZE)
+                {
+                    return new TestAttachmentRequestModel(encodedData, Path.GetFileName(attachment), "", GetAttachmentType(attachment));
+                }
+                else
+                {
+                    _executionContext.Warning(StringUtil.Loc("AttachmentExceededMaximum", attachment));
+                }
+            }
+            else
+            {
+                _executionContext.Warning(StringUtil.Loc("NoSpaceOnDisk", attachment));
+            }
+
+            return null;
+        }
+
+        private TestAttachmentRequestModel GetConsoleLogAttachmentRequestModel(string consoleLog)
+        {
+            if (!string.IsNullOrWhiteSpace(consoleLog))
+            {
+                string consoleLogFileName = "Standard Console Output.log";
+
+                if (consoleLog.Length <= TCM_MAX_FILESIZE)
+                {
+                    byte[] bytes = System.Text.Encoding.UTF8.GetBytes(consoleLog);
+                    string encodedData = Convert.ToBase64String(bytes);
+                    return new TestAttachmentRequestModel(encodedData, consoleLogFileName, "",
+                        AttachmentType.ConsoleLog.ToString());
+                }
+                else
+                {
+                    StringUtil.Loc("AttachmentExceededMaximum", consoleLogFileName);
```

Comment:
Did you mean to throw an exception here?
StringUtil.Loc will only return a localized string, it has to be used by the caller.